### PR TITLE
[vcpkg_configure_make][vcpkg_build_make] Use prefix without DESTDIR

### PR DESCRIPTION
**Pull request by ras0219-msft**
**This PR has been superceded by https://github.com/microsoft/vcpkg/pull/16775 and remains open as a draft for the other improvements to be picked out**

Because vcpkg requires packages to be relocatable, there's no functional difference between `DESTDIR` and `--prefix`. By assigning `--prefix` to `packages\`, we can simplify the build process and avoid doubling path lengths.

### CODE REVIEW

**Comment by JackBoosY** [comment]
cc @Neumann-A for review this PR.

**Comment by ras0219** [comment]
@cenit As part of this, I've run into the non-relocatability of openmpi -- I believe I've built a solution using wrapper scripts, but if you have time I'd appreciate a second opinion.

https://github.com/microsoft/vcpkg/pull/16313/files#diff-f81251de4b55e635fe558a125065447ca8c81e9f5e916f6ff6341a3d9996a082

**Comment by cenit** [comment]
> @cenit As part of this, I've run into the non-relocatability of openmpi -- I believe I've built a solution using wrapper scripts, but if you have time I'd appreciate a second opinion.
> 
> 
> 
> https://github.com/microsoft/vcpkg/pull/16313/files#diff-f81251de4b55e635fe558a125065447ca8c81e9f5e916f6ff6341a3d9996a082

thanks for asking. Tomorrow i will look into it :)

File: ports/openmpi/opal_wrapper.rel.sh [hunk 1]
```diff
@@ -0,0 +1,5 @@
+#!/bin/bash
```
Comment (cenit) [inline ports/openmpi/opal_wrapper.rel.sh:1]: first thing i notice, bash is "too much" on many hpc nodes (and also this shabang is not posix). The script should just require /bin/sh

tomorrow i will test your scripts but the workaround looks ok. 
in any case, you are right, open-mpi was not relocatable. My bad, very sorry for that.

Comment (ras0219) [reply ports/openmpi/opal_wrapper.rel.sh:1]: I started with `/bin/sh`, but I couldn't find documentation for the `exec -a <progname>` style (everyone assumes bash :p).

Comment (cenit) [reply ports/openmpi/opal_wrapper.rel.sh:1]: i can test easily dash, which is not the original sh but is a very nice modern clone ;)

**Comment by JackBoosY** [comment]
This PR has been inactive for a long time, please reopen it if there is any progress.

**Pull request closed without merging**